portfolio/src/main/java/com/google/sps/servlets/InteractiveChartServlet.java

```diff
@@ -10,27 +16,46 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** TODO(briafassler): store data to a database */
+/** Servlet for the interactive pie chart */
 @WebServlet("/interactive-chart")
 public class InteractiveChartServlet extends HttpServlet {
   private Map<String, Integer> emotionVotes = new HashMap<>();
-  private static String EMOTION_DATA_PARAM = "emotion";
+  private static String EMOTION_DATA = "emotion";
+  private static String VOTES_PARAM = "Votes";
+  private static String TIMESTAMP_PARAM = "timestamp";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(emotionVotes);
+    Query votesQuery = new Query(VOTES_PARAM).addSort(TIMESTAMP_PARAM, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(votesQuery);
```

Comment:
is there a reason you're sorting the data?

Reply:
looking back, there is no reason since there doesn't need to be any specific order of the votes. i've now removed it.